File: file_organizer.py
import os
import shutil
import glob

# -------------------------------
# 1️⃣ SET DIRECTORIES
# -------------------------------
source_dir = r"C:/Users/u1163734/OneDrive - IQVIA/Training/Python refresher/1_File organizer/PDF"
destination_dir = r"C:/Users/u1163734/OneDrive - IQVIA/Training/Python refresher/1_File organizer/Moved PDF_Python"

# -------------------------------
# 2️⃣ FIND ALL PDF FILES RECURSIVELY
# -------------------------------
pdf_files = glob.glob(os.path.join(source_dir, "**", "*.pdf"), recursive=True)

# -------------------------------
# 3️⃣ CREATE DESTINATION FOLDER IF MISSING
# -------------------------------
os.makedirs(destination_dir, exist_ok=True)

# -------------------------------
# 4️⃣ COPY PDF FILES
# -------------------------------
for pdf_file in pdf_files:
    shutil.copy2(pdf_file, destination_dir)

# -------------------------------
# 5️⃣ CHECK COPY SUCCESS
# -------------------------------
copied_pdfs = glob.glob(os.path.join(destination_dir, "*.pdf"))
if copied_pdfs:
    print("✅ PDF files copied successfully to the destination folder.")
else:
    print("⚠️ No PDF files found or copied.")

# -------------------------------
# 6️⃣ REMOVE HYPHENS IN FILENAMES
# -------------------------------
for file_path in copied_pdfs:
    dir_name = os.path.dirname(file_path)
    file_name = os.path.basename(file_path)
    new_file_name = file_name.replace("-", " ")
    new_file_path = os.path.join(dir_name, new_file_name)

    if new_file_path != file_path:
        os.rename(file_path, new_file_path)

print("✅ PDF files renamed successfully (hyphens removed).")

# -------------------------------
# 7️⃣ MANUAL CHECK REMINDER
# -------------------------------
print("""
🔍 FINAL CHECKS (manual steps suggested):
1. Review filenames to ensure the last word is complete.
2. Remove 'et al' from filenames if present.
3. Open PDFs to confirm filenames match article titles.
""")

File: file_organizer.R
# Clear global environment
rm(list=ls())

################################CODES START#####################################

######COPY AND PASTE PDFS FROM INDIVIDUAL ENDNOTE SUBFOLDERS TO ONE FILDER######

# Set the source directory where the subfolders with PDF files are located
source_dir <- "C:/Users/u1163734/OneDrive - IQVIA/General - Amgen PH HTA 1/08. HTA dossier development - new format/Amgen Prolia Traveling Library.Data/PDF"

# Set the destination directory to copy the PDF files
destination_dir <- "C:/Users/u1163734/OneDrive - IQVIA/General - Amgen PH HTA 1/08. HTA dossier development - new format/Moved pdf"

# List all PDF files in subfolders
pdf_files <- list.files(path = source_dir, pattern = ".pdf$", recursive = TRUE, 
                        full.names = TRUE)

# Create the destination directory if it doesn't exist
if (!file.exists(destination_dir)) {
  dir.create(destination_dir)
}

# Copy PDF files to the destination directory
file.copy(pdf_files, destination_dir, overwrite = TRUE)

# Check if the files were copied successfully
if (length(list.files(destination_dir, pattern = ".pdf$", 
                      full.names = TRUE)) > 0) {
  cat("PDF files copied successfully to the destination folder.\n")
} else {
  cat("No PDF files found or copied.\n")
}


#################################SORT FILE NAMES################################

######################REMOVE HYPHEN (-) FROM FILE NAMES#########################

# Function to remove hyphens from file names
remove_hyphens <- function(file_path) {
  file_dir <- dirname(file_path)
  file_name <- basename(file_path)
  new_file_name <- gsub("-", " ", file_name)
  new_file_path <- file.path(file_dir, new_file_name)
  file.rename(file_path, new_file_path)
}

# List all PDF files in the folder and subfolders
pdf_files <- list.files(path = destination_dir, pattern = ".pdf$", 
                        recursive = TRUE, full.names = TRUE)

# Rename PDF files to remove hyphens
for (pdf_file in pdf_files) {
  remove_hyphens(pdf_file)
}

# Check if the files were renamed successfully
if (all(file.exists(pdf_files))) {
  cat("PDF files renamed successfully.\n")
} else {
  cat("File renaming failed for some PDF files.\n")
}

##########################FINAL FILE NAME CHECKS################################
# 1. Eyeballing file names to check whether the last word is complete;
#    if not open the file and check and complete/remove the last word.
# 2. Eyeballing file names to look for "et al" and remove them.
# 3. Open the pdfs to check whether the file names match the article titles;
#    if not then search and download the article.

##############################END OF CODES######################################
